pkg/api/persistentvolume/util.go

```diff
@@ -134,11 +132,3 @@ func VisitPVSecretNames(pv *api.PersistentVolume, visitor Visitor) bool {
 	}
 	return true
 }
-
-// DropDisabledAlphaFields removes disabled fields from the pv spec.
-// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pv spec.
-func DropDisabledAlphaFields(pvSpec *api.PersistentVolumeSpec) {
```

Comment:
@liggitt, if a feature goes Beta (enabled by default), shall we still drop its API fields if admin turns it off?

Reply:
Should we remove this in beta?  Or wait until GA?

Reply:
@liggitt when graduating from alpha to beta, do we still need to drop the API field when the feature is disabled? based on https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#alpha-field-in-existing-api-version I assumed field-dropping applies only to alpha, and once in beta we can keep the field populated even if the feature gate is (manually) disabled

Reply:
If the intent of disabling the feature gate is to make the server act as though this field and feature does not exist, these should be kept, at least in some form.

For new data coming in with create API calls, I don't think we'd want fields related to a disabled feature persisted.

For already-persisted data being processed in an update API call, it's less clear what we should do. The scenario is that the feature gate is enabled, some data with this field populated is persisted, then the feature gate is disabled, and the object with the field is updated, keeping the field populated.

For now, keep the feature gate checking/dropping in place. I'm queueing this topic up for a discussion in sig-arch, and will keep you in the loop as guidance here is clarified or changed.

cc @lavalamp

Reply:
I added feature-gates to today's SIG Arch agenda.